src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,151 @@
+package parser;
+
+import java.util.ArrayList;
+
+import exception.DukeException;
+import exception.InvalidInputException;
+import exception.NoTaskDescriptionException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+import tasklist.TaskList;
+import ui.Ui;
+
+/**
+ * Class to encapsulate all parser methods, which parse through user inputs.
+ */
+public class Parser {
+    /**
+     * Returns Task formed from given string.
+     * If input string is not of recognised form, exception is thrown.
+     *
+     * @param echo Input string, treated as commands.
+     * @return Task corresponding to given input commands.
+     * @throws NoTaskDescriptionException If command is given for a task without a description.
+     * @throws InvalidInputException If string does not start with recognised command.
+     */
+    public static Task parseEcho(String echo) throws DukeException {
+        if (echo.startsWith("todo")) {
+            if (echo.substring(4).trim().isEmpty()) {
+                throw(new NoTaskDescriptionException("     ☹ OOPS!!! The description of a todo cannot be empty."));
+            }
+            return new Todo(echo.substring(4).trim());
```

Comment:
`echo.substring(4).trim()` is repeated here, maybe it be put in a variable called "todoArguments" instead which could increase the readability of this section.